mixed-content/generic/common.js

```diff
@@ -0,0 +1,268 @@
+function guid() {
+  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+    var r = Math.random()*16|0, v = c == 'x' ? r : (r&0x3|0x8);
+    return v.toString(16);
+  });
+}
+
+function xhrRequest(url, responseType) {
+  return new Promise(function(resolve, reject) {
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', url, true);
+    xhr.responseType = responseType || "json";
+
+    xhr.addEventListener("error", function() {
+      reject(Error("Network Error"));
+    });
+
+    xhr.addEventListener("load", function() {
+      if (xhr.status != 200)
+        return reject(Error(xhr.statusText));
+
+      resolve(xhr.response);
+    });
+
+    xhr.send();
+  });
+}
+
+function timeoutPromise(timeout) {
+  return new Promise(function(resolve, reject) {
+    setTimeout(resolve, timeout);
+  })
+}
+
+function estimateResponseTime(url, numMeasurements) {
+
+  function measureRoundtripTime(url, index) {
+    return new Promise(function(resolve, reject) {
+      var xhr = new XMLHttpRequest();
+      xhr.open('GET', url);
+      var startTime = (new Date()).getTime();
+
+      xhr.onerror = function() {
+        reject(Error("Network Error"));
+      };
+
+      xhr.onload = function() {
+        if (xhr.status != 200)
+          return reject(Error(xhr.statusText));
+
+        var roundtrip_ms = (new Date()).getTime() - startTime;
+        resolve({roundtrip_ms:roundtrip_ms, index:index});
+      };
+
+      xhr.send();
+    });
+  }
+
+  function roundTo2Decimals(number) {
+    return Math.round(number * 100.0) / 100.0;
+  }
+
+  function getStats(results) {
+    var stats = {
+      sum: 0,
+      min: Number.MAX_VALUE,
+      max: Number.MIN_VALUE,
+      avg: 0,
+      count: 0,
+      stdev: 0,
+    };
+
+    for (var i in results) {
+      stats.sum += results[i];
+      stats.min = Math.min(stats.min, results[i]);
+      stats.max = Math.max(stats.max, results[i]);
+      ++stats.count;
+    }
+
+    // TODO(kristijanburnik): assert count > 0.
+    stats.avg = roundTo2Decimals(stats.sum / stats.count);
+
+    for (var i in results) {
+      var delta = results[i] - stats.avg
+      stats.stdev += delta * delta;
+    }
+
+    stats.stdev /= stats.count;
+    stats.stdev = roundTo2Decimals(Math.sqrt(stats.stdev));
+
+    return stats;
+  }
+
+  return new Promise(function(resolve, reject) {
+    var results = [];
+    var p = measureRoundtripTime(url, 0);
+
+    for (var i = 0; i < numMeasurements - 1; ++i) {
+      p = p.then(function(result) {
+        results.push(result.roundtrip_ms);
+        return measureRoundtripTime(url, result.index + 1);
+      }, reject);
+    }
+
+    p.then(function() {
+      resolve({
+        results: results,
+        stats: getStats(results)
+      })
+    });
+  });
+}
+
+function setAttributes(el, attrs) {
+  attrs = attrs || {}
+  for (var attr in attrs)
+    el.setAttribute(attr, attrs[attr]);
+}
+
+function createElement(tagName, attrs, parent, bindEvents) {
```

Comment:
Passing booleans is tough to read; I'd suggest turning this into an enum-style object thing (e.g. https://developers.google.com/closure/compiler/docs/js-for-compiler#tag-enum).